src/cooldown.rs

```diff
@@ -122,6 +139,24 @@ impl CooldownTracker {
             self.member_invocations.insert((ctx.user_id, guild_id), now);
         }
     }
+    /// Sets the last invocation for the specified cooldown bucket.
+    pub fn set_last_invocation(&mut self, cooldown_type: CooldownType, instant: Instant) {
+        match cooldown_type {
+            CooldownType::Global => self.global_invocation = Some(instant),
+            CooldownType::User(user_id) => {
+                self.user_invocations.insert(user_id, instant);
+            }
+            CooldownType::Guild(guild_id) => {
+                self.guild_invocations.insert(guild_id, instant);
+            }
+            CooldownType::Channel(channel_id) => {
+                self.channel_invocations.insert(channel_id, instant);
+            }
+            CooldownType::Member(member) => {
+                self.member_invocations.insert(member, instant);
+            }
+        }
```

Comment:
Reminds me of the "no boolean parameters" clean code rule, but with 5 execution paths instead of 2... I'd personally have made 5 individual functions instead

But this is nitpicky. I'm fine with this version too

Reply:
This was decided against in the end: https://canary.discord.com/channels/381880193251409931/1246193397950058809/1246385743144681502